qfieldsync/gui/map_layer_config_widget.py

```diff
@@ -70,15 +73,31 @@ def __init__(self, layer, canvas, parent):
 
         self.isGeometryLockedCheckBox.setEnabled(self.layer_source.can_lock_geometry)
         self.isGeometryLockedCheckBox.setChecked(self.layer_source.is_geometry_locked)
+
         self.photoNamingTable = PhotoNamingTableWidget()
         self.photoNamingTable.addLayerFields(self.layer_source)
         self.photoNamingTable.setLayerColumnHidden(True)
 
-        # insert the table as a second row only for vector layers
+        # append the photo naming table to the layout
         if Qgis.QGIS_VERSION_INT >= 31300 and layer.type() == QgsMapLayer.VectorLayer:
-            self.layout().insertRow(2, self.tr("Photo naming"), self.photoNamingTable)
+            self.layout().insertRow(-1, self.tr("Photo naming"), self.photoNamingTable)
             self.photoNamingTable.setEnabled(self.photoNamingTable.rowCount() > 0)
 
+        self.relationshipConfigurationTable = RelationshipConfigurationTableWidget()
+        self.relationshipConfigurationTable.addLayerFields(self.layer_source)
+        self.relationshipConfigurationTable.setLayerColumnHidden(True)
+
+        # append the relationship configuration table to the layout
+        if Qgis.QGIS_VERSION_INT >= 31300 and layer.type() == QgsMapLayer.VectorLayer:
```

Comment:
I think we can forget about QGIS 3.14 and simplify the check here and on line 82.